### DISCUSSION TITLE
Formula php does not create the correct ext_dir for pecl

### DISCUSSION THREAD

**Discussion by raumi75** [Homebrew/discussions#5053] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 6 days ago
Core tap JSON: 12 Jan 21:26 UTC
Core cask tap JSON: 12 Jan 21:26 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: emacsclient -t
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I installed php with  `brew install php`, which works great.

Then I wanted to install XDebug using `arch -arm64 sudo pecl install xdebug` which failed.

This is the relevant output where it failed.

```
Build process completed successfully
Installing '/opt/homebrew/Cellar/php/8.3.1/pecl/20230831/xdebug.so'

Warning: mkdir(): File exists in System.php on line 294
PHP Warning:  mkdir(): File exists in /opt/homebrew/Cellar/php/8.3.1/share/php/pear/System.php on line 294

Warning: mkdir(): File exists in /opt/homebrew/Cellar/php/8.3.1/share/php/pear/System.php on line 294
ERROR: failed to mkdir /opt/homebrew/Cellar/php/8.3.1/pecl/20230831
```

The reason appears to be this:

```
% pecl config-get ext_dir
/opt/homebrew/lib/php/pecl/20230831
```

After creating the directory with `mkdir -p /opt/homebrew/lib/php/pecl/20230831`, the pecl command worked and XDebug was installed.

 I tried to check the code in the [php formula](https://github.com/Homebrew/homebrew-core/blob/af6a4cefb208196759cd6d33c1db83c6172a3999/Formula/p/php.rb) and it looked like it should create all the relevant directories. No idea why it did not for pecl.

I hope this description helps and someone with more experience can fix the formula, so others don't run into the same issue.

**Comment by hamstu** [comment]
Thanks @raumi75! This worked for me. 🙏🏻 

<img src="https://media1.giphy.com/media/ZfK4cXKJTTay1Ava29/giphy.gif"/>

**Comment by Satyabrat35** [comment]
Thank You!

**Comment by lchhieu** [comment]
Excellent. I have an issue with install imagick and following your solution I can installed it

**Comment by AragornYY** [comment]
The issue is on PECL side. According to the error, PECL failed to mkdir /opt/homebrew/Cellar/php/8.3.1/pecl/20230831. This path appears to be derived from the path of the pecl binary.

However, PECL should respect the configuration shown below:
```bash
% pecl config-get ext_dir
/opt/homebrew/lib/php/pecl/20230831
```
